### PR TITLE
KFSPTS-11963: Add PMW column for supplier upload status

### PR DESCRIPTION
**Pull request by cah292**
This is not quite ready to be merged yet since I'm still validating locally. However, I wanted to go ahead and open up a PR for this, in case someone has cycles to review this to make sure it's meeting the expected functionality.

### CODE REVIEW

**Comment by cah292** [comment]
This is now ready to be merged once it has been reviewed and approved.

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksDtoToPaymentWorksVendorConversionServiceImpl.java [hunk 1]
```diff
@@ -46,6 +46,7 @@ public PaymentWorksVendor createPaymentWorksVendorFromPaymentWorksNewVendorReque
         PaymentWorksVendor stgVendor = new PaymentWorksVendor();
         if (newVendorDetailExists(pmwNewVendorRequestDetailDTO)) {
             stgVendor.setPmwVendorRequestId(pmwNewVendorRequestDetailDTO.getId());
+            stgVendor.setSupplierUploadStatus(PaymentWorksConstants.SupplierUploadStatus.PENDING_PAAT);
             populateNewVendorRequestingCompanyAttributes(stgVendor, pmwNewVendorRequestDetailDTO);
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksDtoToPaymentWorksVendorConversionServiceImpl.java]: Same comment as above. IMHO, I do not think status should not be set here.

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java [hunk 2]
```diff
@@ -183,7 +183,7 @@ public void registerKfsPvenApprovalForExistingPaymentWorksVendor(String kfsVendo
                      + "' and kfsVendorDocumentNumber '" + kfsVendorDocumentNumber + "' and pmwVendorRequestId '" + pmwVendor.getPmwVendorRequestId()
                      + "' and vendor number '");
             getPaymentWorksVendorDao().updateExistingPaymentWorksVendorInStagingTable(pmwVendor.getId(), PaymentWorksConstants.KFSVendorProcessingStatus.VENDOR_APPROVED, vendorDetail.getVendorHeaderGeneratedIdentifier(),
-                                                                                      vendorDetail.getVendorDetailAssignedIdentifier(), getDateTimeService().getCurrentTimestamp());
+                                                                                      vendorDetail.getVendorDetailAssignedIdentifier(), PaymentWorksConstants.SupplierUploadStatus.PENDING_PAAT, getDateTimeService().getCurrentTimestamp());
         } else {
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: For consistency with the way that the other status values are set, I do not believe that the SupplierUploadStatus should be set here. It is appropriately set in service class PaymentWorksNewVendorPayeeAchServiceImpl in method setStatusValuesForPaymentWorksProcessedNewVendorKfsVendorCreatedAchPendingPven and the upload back to PMW cannot happen for a vendor originating in PMW cannot happen until the PAAT batch job runs.

Comment (cah292) [reply src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: Ok, I'll remove the status setup here.

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksDtoToPaymentWorksVendorConversionServiceImpl.java (hunk 1 above)
Comment (cah292) [reply src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksDtoToPaymentWorksVendorConversionServiceImpl.java]: Ok, I'll remove the status setup here.

**Comment by cah292** [comment]
I've committed some updates based on the PR feedback, and have done some local validation to verify that things are still working. This is ready for review again.

**Comment by cah292** [comment]
FYI, I think I still need to change the registerKfsPvenDisapprovalForExistingPaymentWorksVendor() method, in order for it to be consistent with the other changes you had suggested. I still need to validate this quick change before I commit it.

File: src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/PaymentWorksVendorDao.java [hunk 3]
```diff
@@ -4,14 +4,14 @@
 
 public interface PaymentWorksVendorDao {
 
-    void updateExistingPaymentWorksVendorInStagingTable(Integer id, String pmwRequestStatus, String kfsVendorProcessingStatus, Timestamp processingTimeStamp);
+    void updateExistingPaymentWorksVendorInStagingTable(Integer id, String pmwRequestStatus, String kfsVendorProcessingStatus, String supplierUploadStatus, Timestamp processingTimeStamp);
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/PaymentWorksVendorDao.java]: This method signature should not be changed

File: src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/PaymentWorksVendorDao.java [hunk 4]
```diff
@@ -4,14 +4,14 @@
...
 
-    void updateExistingPaymentWorksVendorInStagingTable(Integer id, String pmwRequestStatus, String kfsVendorProcessingStatus, String kfsAchProcessingStatus, String kfsVendorDocumentNumber, Timestamp processingTimeStamp);
+    void updateExistingPaymentWorksVendorInStagingTable(Integer id, String pmwRequestStatus, String kfsVendorProcessingStatus, String kfsAchProcessingStatus, String kfsVendorDocumentNumber, String supplierUploadStatus, Timestamp processingTimeStamp);
 
-    void updateExistingPaymentWorksVendorInStagingTable(Integer id, String kfsVendorProcessingStatus, Integer vendorHeaderGeneratedIdentifier, Integer vendorDetailAssignedIdentifier, Timestamp processingTimeStamp);
+    void updateExistingPaymentWorksVendorInStagingTable(Integer id, String kfsVendorProcessingStatus, Integer vendorHeaderGeneratedIdentifier, Integer vendorDetailAssignedIdentifier, String supplierUploadStatus, Timestamp processingTimeStamp);
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/PaymentWorksVendorDao.java]: This method signature should not be changed.

File: src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/impl/PaymentWorksVendorDaoJdbc.java [hunk 5]
```diff
@@ -20,41 +19,41 @@
...
     }
     
     @Override
-    public void updateExistingPaymentWorksVendorInStagingTable(Integer id, String kfsVendorProcessingStatus, Integer vendorHeaderGeneratedIdentifier, Integer vendorDetailAssignedIdentifier, Timestamp processingTimeStamp) {
-        String updateSql = buildUpdateExistingPaymentWorksVendorInStagingTableSql(id, KFSConstants.EMPTY_STRING, kfsVendorProcessingStatus, KFSConstants.EMPTY_STRING, KFSConstants.EMPTY_STRING, processingTimeStamp, vendorHeaderGeneratedIdentifier, vendorDetailAssignedIdentifier, null);
+    public void updateExistingPaymentWorksVendorInStagingTable(Integer id, String kfsVendorProcessingStatus, Integer vendorHeaderGeneratedIdentifier, Integer vendorDetailAssignedIdentifier, String supplierUploadStatus, Timestamp processingTimeStamp) {
+        String updateSql = buildUpdateExistingPaymentWorksVendorInStagingTableSql(id, KFSConstants.EMPTY_STRING, kfsVendorProcessingStatus, KFSConstants.EMPTY_STRING, KFSConstants.EMPTY_STRING, supplierUploadStatus, processingTimeStamp, vendorHeaderGeneratedIdentifier, vendorDetailAssignedIdentifier, null);
         LOG.info("updateExistingPaymentWorksVendorInStagingTable: updateSQL = " + updateSql);
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/impl/PaymentWorksVendorDaoJdbc.java]: Method signature should not be changed and KFSConstants.EMPTY_STRING should be used in buildUpdateExistingPaymentWorksVendorInStagingTableSql call.

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java [hunk 6]
```diff
@@ -211,7 +211,8 @@ public void registerKfsPvenDisapprovalForExistingPaymentWorksVendor(String kfsVe
             LOG.info("registerKfsPvenDisapprovalForExistingPaymentWorksVendor: Disapproving PaymentWorks originating Vendor with table ID '" + pmwVendor.getId()
                      + "' and kfsVendorDocumentNumber '" + kfsVendorDocumentNumber + "' and pmwVendorRequestId '" + pmwVendor.getPmwVendorRequestId() + "'");
             getPaymentWorksVendorDao().updateExistingPaymentWorksVendorInStagingTable(pmwVendor.getId(), PaymentWorksConstants.KFSVendorProcessingStatus.VENDOR_DISAPPROVED, vendorDetail.getVendorHeaderGeneratedIdentifier(),
-                                                                                      vendorDetail.getVendorDetailAssignedIdentifier(), getDateTimeService().getCurrentTimestamp());
+                                                                                      vendorDetail.getVendorDetailAssignedIdentifier(), PaymentWorksConstants.SupplierUploadStatus.INELIGIBLE_FOR_UPLOAD,
+                                                                                      getDateTimeService().getCurrentTimestamp());
         } else {
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: As you previously noted, PaymentWorksConstants.SupplierUploadStatus.INELIGIBLE_FOR_UPLOAD should be removed.

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java [hunk 7]
```diff
@@ -183,7 +183,7 @@ public void registerKfsPvenApprovalForExistingPaymentWorksVendor(String kfsVendo
                      + "' and kfsVendorDocumentNumber '" + kfsVendorDocumentNumber + "' and pmwVendorRequestId '" + pmwVendor.getPmwVendorRequestId()
                      + "' and vendor number '");
             getPaymentWorksVendorDao().updateExistingPaymentWorksVendorInStagingTable(pmwVendor.getId(), PaymentWorksConstants.KFSVendorProcessingStatus.VENDOR_APPROVED, vendorDetail.getVendorHeaderGeneratedIdentifier(),
-                                                                                      vendorDetail.getVendorDetailAssignedIdentifier(), getDateTimeService().getCurrentTimestamp());
+                                                                                      vendorDetail.getVendorDetailAssignedIdentifier(), KFSConstants.EMPTY_STRING, getDateTimeService().getCurrentTimestamp());
         } else {
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: KFSConstants.EMPTY_STRING value should be removed.

**Comment by cah292** [comment]
I've made more changes based on the latest feedback, and things still appear to be working in my local environment. This is ready for review again.

**Pull request merged**